pkg/cache/warm.go

```diff
@@ -152,13 +153,27 @@ func (w *Warmer) Warm(image string, opts *config.WarmerOptions) (v1.Hash, error)
 		return v1.Hash{}, errors.Wrapf(err, "Failed to retrieve digest: %s", image)
 	}
 
+TRY_LOCAL:
```

Comment:
It is not recommended to use GOTO statements as it makes the code hard to follow and understand. In this situation, it would be more idiomatic in Go to use a for or while loop to retry the lock acquisition.  Can you change this to use a retry loop instead?  Additionally it likely makes sense to cap the amount of time the process can wait to acquire the file lock so that it doesn't spin forever

Reply:
I removed the GOTO and using for loop now. 

I think it's okay not to set the maximum retry limits for the for loop here.
I designed the renewal and timeout of the lock, so there should be no deadlock here. If the holder of lock was died and cannot be renewed, other processes will have the right to unlock. And locking will continue through the entire download process, which may be very long and difficult to provide a suitable default value for timeout.